packages/@aws-cdk/aws-kinesisfirehose/lib/processor.ts

```diff
@@ -0,0 +1,96 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { Duration, Size } from '@aws-cdk/core';
+
+import { Construct } from 'constructs';
+
+/**
+ * Configure the data processor.
+ */
+export interface DataProcessorProps {
+  /**
+   * The length of time Kinesis Data Firehose will buffer incoming data before calling the processor.
+   *s
+   * @default Duration.minutes(1)
+   */
+  readonly bufferInterval?: Duration;
+
+  /**
+   * The amount of incoming data Kinesis Data Firehose will buffer before calling the processor.
+   *
+   * @default Size.mebibytes(3)
+   */
+  readonly bufferSize?: Size;
+
+  /**
+   * The number of times Kinesis Data Firehose will retry the processor invocation after a failure due to network timeout or invocation limits.
+   *
+   * @default 3
+   */
+  readonly retries?: number;
+}
+
+/**
+ * The key-value pair that identifies the underlying processor resource.
+ *
+ * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-kinesisfirehose-deliverystream-processorparameter.html
+ */
+export interface DataProcessorIdentifier {
+  /**
+   * The parameter name that corresponds to the processor resource's identifier.
+   *
+   * Must be an accepted value in `CfnDeliveryStream.ProcessoryParameterProperty.ParameterName`.
+   */
+  readonly parameterName: string;
+
+  /**
+   * The identifier of the underlying processor resource.
+   */
+  readonly parameterValue: string;
+}
+
+/**
+ * The full configuration of a data processor.
+ */
+export interface DataProcessorConfig {
+  /**
+   * The type of the underlying processor resource.
+   *
+   * Must be an accepted value in `CfnDeliveryStream.ProcessorProperty.Type`.
+   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-kinesisfirehose-deliverystream-processor.html#cfn-kinesisfirehose-deliverystream-processor-type
+   * @example 'Lambda'
+   */
+  readonly processorType: string;
+
+  /**
+   * The key-value pair that identifies the underlying processor resource.
+   */
+  readonly processorIdentifier: DataProcessorIdentifier;
+}
+
+/**
+ * Options when binding a DataProcessor to a delivery stream destination.
+ */
+export interface DataProcessorBindOptions {
+  /**
+   * The IAM role assumed by Kinesis Data Firehose to write to the destination that this DataProcessor will bind to.
+   */
+  readonly role: iam.IRole;
+}
+
+/**
+ * A data processor that Kinesis Data Firehose will call to transform records before delivering data.
+ */
+export interface IDataProcessor {
+  /**
+   * The constructor props of the DataProcessor.
+   */
+  readonly props: DataProcessorProps;
```

Comment:
this feels weird to me, I think the information here should be returned from `bind`. it's fine if we don't have DataProcessorConfig extend DataProcessorProps, can just replicate the properties

Reply:
I don't think we should replicate the properties if we can help that.

Reply:
I don't think we should be returning the creation properties in `bind()` - that's just a weird abstraction.

`bind()` is properties specific to a given resource pairing (this, the thing that calls `bind()` on it) - it shouldn't contain properties that are common to all instances of this thing.

Reply:
Also these props are not something new created by the `bind` or added to a new object created by the `bind`. They are just each referenced one-by-one in the function that calls `bind`. So I think it is fine to make the `props` a part of the `LambdaFunctionProcessor` itself.

Reply:
I guess we could consider moving this logic: https://github.com/aws/aws-cdk/pull/15704/files#diff-855c52287d23616907e597804078115aaab493e99046c7940761585890f1d14cR108-R120 

inside of the `bind`.

Reply:
@BenChaimberg , take a look at this commit: https://github.com/aws/aws-cdk/pull/15704/commits/ca59756878e42255a229fe465933f1c3477694e6 

I think I want to revert it because once other kinds of data processors are available besides Lambda, that future change would require more refactoring.